third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -400,32 +447,41 @@ impl<'env> Generator<'env> {
                         ),
                     );
                 }
-                self.emit_call(*id, targets, BytecodeOperation::WriteRef, vec![
-                    lhs_temp, rhs_temp,
-                ])
+                let src_ids = vec![lhs.node_id(), rhs.node_id()];
+                self.emit_call(
+                    *id,
+                    targets,
+                    target_node_ids,
+                    BytecodeOperation::WriteRef,
+                    vec![lhs_temp, rhs_temp],
+                    src_ids,
+                )
             },
             ExpData::Assign(id, lhs, rhs) => self.gen_assign(*id, lhs, rhs, None),
             ExpData::Return(id, exp) => {
                 let results = self.results.clone();
-                self.gen(results.clone(), exp);
-                self.emit_with(*id, |attr| Bytecode::Ret(attr, results))
+                let results_ids = vec![exp.node_id(); results.len()];
+                self.gen(results.clone(), results_ids.clone(), exp);
+                self.emit_with(*id, Vec::new(), results_ids, |attr| {
+                    Bytecode::Ret(attr, results)
+                })
             },
             ExpData::IfElse(id, cond, then_exp, else_exp) => {
                 let cond_temp = self.gen_escape_auto_ref_arg(cond, false);
                 let then_label = self.new_label(*id);
                 let else_label = self.new_label(*id);
                 let end_label = self.new_label(*id);
-                self.emit_with(*id, |attr| {
+                self.emit_with(*id, Vec::new(), vec![cond.node_id()], |attr| {
                     Bytecode::Branch(attr, then_label, else_label, cond_temp)
                 });
                 let then_id = then_exp.node_id();
-                self.emit_with(then_id, |attr| Bytecode::Label(attr, then_label));
-                self.gen(targets.clone(), then_exp);
-                self.emit_with(then_id, |attr| Bytecode::Jump(attr, end_label));
+                self.emit_without_target_src_ids(then_id, |attr| Bytecode::Label(attr, then_label));
```

Comment:
This name is pretty verbose.  Can we maybe call it `emit_with_no_target` instead?

Reply:
done